File: nfclib/src/main/java/com/ppy/nfclib/INfcCardReader.java
package com.ppy.nfclib;

import android.content.Intent;

import java.io.IOException;

/**
 * NFC调用相关接口.
 * Created by ZP on 2017/9/20.
 */

public interface INfcCardReader {

    public void onCreate(Intent intent);

    public void onStart();

    public void onResume();

    public void onPause();

    public void onStop();

    public void onDestroy();

    public void onNewIntent(Intent intent);

    public String sendData(byte[] data) throws IOException;

    public String sendData(String hexData) throws IOException;

    public byte[] tranceive(byte[] data) throws IOException;

    public byte[] tranceive(String hexData) throws IOException;

    public boolean isCardConnected();

}

File: nfclib/src/main/java/com/ppy/nfclib/INfcCardReader.kt
package com.ppy.nfclib

import android.content.Intent

import java.io.IOException

/**
 * NFC调用相关接口.
 * Created by ZP on 2017/9/20.
 */

interface INfcCardReader {

    fun isCardConnect(): Boolean

    fun onCreate(intent: Intent)

    fun onStart()

    fun onResume()

    fun onPause()

    fun onStop()

    fun onDestroy()

    fun onNewIntent(intent: Intent)

    @Throws(IOException::class)
    fun sendData(data: ByteArray): String

    @Throws(IOException::class)
    fun sendData(hexData: String): String

    @Throws(IOException::class)
    fun tranceive(data: ByteArray): ByteArray

    @Throws(IOException::class)
    fun tranceive(hexData: String): ByteArray

}
